### PR TITLE
Modify MappingModelCoverageAnalysis to return PMCD for the model coverage

### PR DESCRIPTION
**Pull request by gayathrir11**
#### What type of PR is this?
Improvement

#### Which issue(s) this PR fixes:

Related to finos/legend-studio#2346

### CODE REVIEW

File: legend-engine-xts-analytics/legend-engine-xts-analytics-search/legend-engine-xt-analytics-search-pure/src/main/resources/core_analytics_search/trans.pure [hunk 1]
```diff
@@ -114,7 +114,7 @@ function meta::analytics::search::transformation::dataspace::buildExecutionConte
 
 function meta::analytics::search::transformation::dataspace::buildClassDocuments(mapping: meta::pure::mapping::Mapping[1]): meta::analytics::search::metamodel::class::SimpleClassElement[*]
 {
-  let analysedMapping = meta::analytics::mapping::modelCoverage::analyze($mapping, true, true);
+  let analysedMapping = meta::analytics::mapping::modelCoverage::analyze($mapping, true, true, false);
```
Comment (akphi) [inline legend-engine-xts-analytics/legend-engine-xts-analytics-search/legend-engine-xt-analytics-search-pure/src/main/resources/core_analytics_search/trans.pure:117]: why do we do this?

File: legend-engine-xts-analytics/legend-engine-xts-analytics-mapping/legend-engine-xt-analytics-mapping-pure/src/main/resources/core_analytics_mapping/modelCoverage/serializer.pure [hunk 2]
```diff
@@ -49,6 +49,7 @@ function <<access.private>> meta::analytics::mapping::modelCoverage::serializati
           | $entity.info->toOne()->mappedEntityInfoToJSON()
     )),
     newJSONKeyValue('path', ^JSONString(value=$entity.path)),
+    newJSONKeyValue('classPath', ^JSONString(value=$entity.classPath)),
```
Comment (akphi) [inline legend-engine-xts-analytics/legend-engine-xts-analytics-mapping/legend-engine-xt-analytics-mapping-pure/src/main/resources/core_analytics_mapping/modelCoverage/serializer.pure:52]: should we put this in `info`?

File: legend-engine-xts-analytics/legend-engine-xts-analytics-mapping/legend-engine-xt-analytics-mapping-pure/src/main/resources/core_analytics_mapping/modelCoverage/utility.pure [hunk 3]
```diff
@@ -60,28 +60,6 @@ function meta::analytics::mapping::modelCoverage::utility::getFunctionPath(f:Fun
    $f->elementToPath()->replace($f.name->toOne(), $f.functionName->toOne());
 }
 
-
-function {doc.doc = 'Get the parent or super types of the provided class'}
```
Comment (akphi) [inline legend-engine-xts-analytics/legend-engine-xts-analytics-mapping/legend-engine-xt-analytics-mapping-pure/src/main/resources/core_analytics_mapping/modelCoverage/utility.pure:64]: you're sure nothing is using these functions?

**Review by akphi — CHANGES_REQUESTED** [review]
You might want to rebase and update snapshot version number

File: legend-engine-xts-analytics/legend-engine-xts-analytics-mapping/legend-engine-xt-analytics-mapping-pure/src/main/resources/core_analytics_mapping/modelCoverage/utility.pure (hunk 3 above)
Comment (gayathrir11) [reply legend-engine-xts-analytics/legend-engine-xts-analytics-mapping/legend-engine-xt-analytics-mapping-pure/src/main/resources/core_analytics_mapping/modelCoverage/utility.pure:64]: We are using this functions. Just moved it to different place. I moved it to class analytics utility

File: legend-engine-xts-analytics/legend-engine-xts-analytics-search/legend-engine-xt-analytics-search-pure/src/main/resources/core_analytics_search/trans.pure (hunk 1 above)
Comment (gayathrir11) [reply legend-engine-xts-analytics/legend-engine-xts-analytics-search/legend-engine-xt-analytics-search-pure/src/main/resources/core_analytics_search/trans.pure:117]: I added a new boolean to mapping analysis whether to analyze the mapped classes or not

**Review by akphi — DISMISSED** [review]

**Review by akphi — APPROVED** [review]

**Review by akphi — APPROVED** [review]

**Review by gs-ssh16 — APPROVED** [review]
Approving change in PureIDELight file

**Review by gs-ssh16 — DISMISSED** [review]

**Review by gs-ssh16 — APPROVED** [review]

**Pull request merged**